btconfig/utils/dataloader/ib.py

```diff
@@ -163,8 +168,26 @@ def _processRequests(self):
                 whatToShow=str(request['what']),
                 useRTH=int(request['useRTH']),
                 formatDate=1)
+
             while r in self._requests and self.engine.isConnected():
+
```

Comment:
no need to do anything while waiting for request  to finish, the request is finished when while loop exits.